app/models/enterprise.rb

```diff
@@ -247,6 +247,13 @@ def self.distinct_count
     count(distinct: true)
   end
 
+  # Remove any unsupported HTML.
+  def long_description=(html)
+    super(HtmlSanitizer.sanitize(
+      html, tags: %w[h1 h2 h3 h4 p b i u a], attributes: %w[href target],
```

Comment:
I noticed that Trix supports some other tags, and actually uses `strong` and `em` instead of `b` and `i`. Ugh, and so many `br`s:

![Screenshot 2024-05-13 at 11 18 09 am](https://github.com/openfoodfoundation/openfoodnetwork/assets/4188088/b8ad93d9-871d-49b4-bc25-8e7c655ee4cd)

https://staging.openfoodnetwork.org.au/elmore-compost-and-organics/shop
```html
<p class="text-small ng-binding" data-controller="add-blank-to-link" ng-bind-html="::product.description_html">
<div>
   Biochar is a porous charcoal well recognised for being able to hold water, create great building blocks for biology and a store of plant carbon.<br><br><br>
   <strong>
   bold
   </strong>
   <br>
   <em>italic</em>
   <br>
   <del>
   strikethrough
   </del>
   <br>
   <a href="https://staging.openfoodnetwork.org.au/" target="_blank">
   link
   </a>
   </br>
   </br>
   </br>
   </br>
   </br>
   </br>
</div>
<h1>
   heading
</h1>
<blockquote>
   quote
</blockquote>
<pre>code</pre>
<ul>
   <li>
      bullet list
      <ul>
         <li>
            indented
         </li>
      </ul>
   </li>
</ul>
<ol>
   <li>
      number list
      <ol>
         <li>
            indented
         </li>
      </ol>
   </li>
</ol>
<div>
   <br>
   horizontal rule:
   <hr>
   <br>
   Pasted tables get reformatted by Trix I guess:
   <br>
   <br>
   <strong>
   CompanyContactCountry
   </strong>
   <br>
   Alfreds Futterkiste | Maria Anders | Germany
   <br>
   Centro comercial Moctezuma | Francisco Chang | Mexico
   <br>
   Ernst Handel | Roland Mendel | Austria
   <br>
   Island Trading | Helen Bennett | UK
   <br>
   Laughing Bacchus Winecellars | Yoshi Tannamuri | Canada
   <br>
   Magazzini Alimentari Riuniti | Giovanni Rovelli | Italy
   </br>
   </br>
   </br>
   </br>
   </br>
   </br>
   </br>
   </br>
   </br>
   </hr>
   </br>
</div>
</p>
```

I would err on the side of being more permissive (of safe elements) and allow the other tags in case Trix allows pasting of other html, or we use a different editor one day. Is there a comprehensive list we could copy from somewhere? If not, can we add all the above examples?

Reply:
Ahh, I just realised that the Enterprise description uses a different editor! 
So your list here should be enough. However if we extend this to products we'd need to expand the list.

Reply:
>  Is there a comprehensive list we could copy from somewhere?

Well, when we omit our list of safe tags then Rails comes with a comprehensive list already. We could use that to allow more pasted HTML. But then we have a discrepancy of pasted vs edited. And when it's about security, I usually err on the side of caution and be more restrictive.

Reply:
Here's the full standard list: https://github.com/flavorjones/loofah/blob/main/lib/loofah/html5/safelist.rb

It would allow certain CSS as well. I don't think it's all safe in our case, it could mess up the layout or maybe more.